File: reverseWords.js
// Given a string s, reverse the string without reversing its individual words. Words are separated by dots(.).

// Note: The string may contain leading or trailing dots(.) or multiple dots(.) between two words. The returned string should only have a single dot(.) separating the words, and no extra dots should be included.

// Examples :

// Input: s = "i.like.this.program.very.much"
// Output: "much.very.program.this.like.i"
// Explanation: The words in the input string are reversed while maintaining the dots as separators, resulting in "much.very.program.this.like.i".

class Solution {
    reverseWords(s) {
        // code here
        console.log(s.split('.'));
        let b = s.split('.').filter(Boolean)
        console.log(this.reverseArray(b).join("."));

        
    }
    reverseArray(a){
        let left =0;
        let right = parseInt(a.length) - 1 ;
        let temp;
        while(left < right){
            temp = a[left];
            a[left] = a[right];
            a[right] = temp;
            left ++;
            right -- ;
        }
        return a;
    }
}
const myInstance = new Solution();
myInstance.reverseWords("i..like.this.program.very.much");

File: reverse_words.py
# Given a string s, reverse the string without reversing its individual words.
# Words are separated by dots(.). Handle multiple dots and leading/trailing dots.

class Solution:
    def reverse_words(self, s: str):
        # Split by dots and filter out empty strings (handles multiple dots)
        words = [word for word in s.split('.') if word]
        
        # Reverse the array of words
        self.reverse_array(words)
        
        # Join words with single dots
        return '.'.join(words)
    
    def reverse_array(self, words: list):
        left = 0
        right = len(words) - 1
        
        while left < right:
            # Swap elements
            words[left], words[right] = words[right], words[left]
            left += 1
            right -= 1

# Test cases
if __name__ == "__main__":
    solution = Solution()
    test_cases = [
        "i.like.this.program.very.much",
        "i..like.this.program.very.much",
        "hello...world",
        "python..is.awesome..",
        "..single.word.."
    ]
    
    for test in test_cases:
        print(f"Input:  {test}")
        print(f"Output: {solution.reverse_words(test)}\n")